src/els_text_synchronization.erl

```diff
@@ -42,4 +45,38 @@ generate_diagnostics(Uri) ->
   Params1  = #{ uri => Uri
               , diagnostics => CDiagnostics ++ DDiagnostics ++ EDiagnostics
               },
+  maybe_compile_and_load(Uri, CDiagnostics),
   els_server:send_notification(Method, Params1).
+
+-spec maybe_compile_and_load(uri(), [diagnostic()]) -> ok.
+maybe_compile_and_load(Uri, CDiagnostics) ->
```

Comment:
You can have a first clause where you match `CDiagnostics` with a `[]` to avoid the tuple in the case.